Sources/AsyncKit/ConnectionPool/EventLoopConnectionPool.swift

```diff
@@ -65,26 +74,34 @@ public final class EventLoopConnectionPool<Source> where Source: ConnectionPoolS
     ///   - source: Creates new connections when needed.
     ///   - maxConnections: Limits the number of connections that can be open. Defaults to 1.
     ///   - requestTimeout: Timeout for requesting a new connection. Defaults to 10 seconds.
+    ///   - pruneInterval: How often to check for and prune idle database connections. If `nil` (the default),
+    ///     no pruning is performed.
+    ///   - maxIdleTimeBeforePruning: How long a connection may remain idle before being pruned, if pruning is enabled.
+    ///     Defaults to 2 minutes. Ignored if `pruneInterval` is `nil`.
     ///   - logger: For lifecycle logs.
     ///   - on: Event loop.
     public init(
         source: Source,
         maxConnections: Int,
         requestTimeout: TimeAmount = .seconds(10),
+        pruneInterval: TimeAmount? = nil,
+        maxIdleTimeBeforePruning: TimeAmount = .seconds(120),
```

Comment:
Hmm I'd create a "PruningConfiguration" which can be optional itself, but if provided, has both pruneInterval and maxIdle as non-optional.